apps/src/lib/node/ledger/ethereum_node/oracle.rs

```diff
@@ -127,100 +152,189 @@ pub fn run_oracle(
 /// It also checks that once the specified number of confirmations
 /// is reached, an event is forwarded to the ledger process
 async fn run_oracle_aux(oracle: Oracle) {
-    // Initialize our local state. This includes
-    // the latest block height seen and a queue of events
-    // awaiting a certain number of confirmations
+    // Initialize a queue to keep events which are awaiting a certain number of
+    // confirmations
     let mut pending: Vec<PendingEvent> = Vec::new();
-    const SLEEP_DUR: std::time::Duration = std::time::Duration::from_secs(1);
+
+    // TODO(namada#560): get the appropriate Ethereum block height to start
+    // checking from rather than starting from zero every time
+    let mut next_block_to_process: Uint256 = 0u8.into();
+
     loop {
-        tokio::time::sleep(SLEEP_DUR).await;
-        // update the latest block height
-        let latest_block = loop {
-            match oracle.eth_block_number().await {
-                Ok(height) => break height,
-                Err(error) => {
-                    tracing::warn!(
+        tracing::info!(
+            ?next_block_to_process,
+            "Checking Ethereum block for bridge events"
+        );
+        tokio::select! {
+            result = process(&oracle, &mut pending, next_block_to_process.clone()) => {
```

Comment:
this `tokio::select!` might be unsafe, https://carllerche.com/2021/06/17/six-ways-to-make-async-rust-easier/